libs/langchain-google-cloud-sql-pg/src/indexes.ts

```diff
@@ -0,0 +1,35 @@
+class StrategyMixin {
+  operator: string;
+
+  searchFunction: string;
+
+  indexFunction: string;
+
+  constructor(operator: string, searchFunction: string, indexFunction: string) {
+    this.operator = operator;
+    this.searchFunction = searchFunction;
+    this.indexFunction = indexFunction;
+  }
+}
+
+/**
+ * Enumerator of the Distance strategies.
+ */
+export class DistanceStrategy extends StrategyMixin {
```

Comment:
I'm a bit confused by this pattern - why have `DistanceStrategy` itself extend the mixin?

Reply:
This allows us to create an enum like object where we can really easily call the get the different attributes